### PR TITLE
Run game script

### PR DESCRIPTION
**Pull request by Grant-Postma**
Creating the run and rebuild script.
Checking a handful of files and directory, displaying the error to the user if they are missing.

### CODE REVIEW

File: demos/TicTacToe/rebuild_demo.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  """The main function."""
+  # The run_demo.py script directory.
+  ROOT_DIRECTORY = os.path.dirname(os.path.abspath(__file__))
+  
+  # Sets up the logging format and handler.
+  logger = logger_setup()
+ 
+  # Directory paths.
+  game_name = "tic_tac_toe_demo"
+  build_dir = os.path.join(ROOT_DIRECTORY,game_name,"build")
```
Comment (alexames) [inline demos/TicTacToe/rebuild_demo.py:54]: Make sure spacing is consistent

Comment (Grant-Postma) [reply demos/TicTacToe/rebuild_demo.py:54]: Clang format did c++ has spoiled me. I'll have to look for the equivalent for python or just do a thorough manual check on the three python scripts.

File: demos/TicTacToe/rebuild_demo.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  game_name = "tic_tac_toe_demo"
+  build_dir = os.path.join(ROOT_DIRECTORY, game_name, "build")
+
+  # Checks whether the build directory was created.
+  if os.path.isdir(build_dir):
+    logger.info("Building the demo...")
+    # Builds the tic_tac_toe_demo executable.
+    log_run(build_dir, logger, "cmake --build .")
+  else:
+    logger.error("Build directory does not exist.")
```
Comment (DellaBitta) [inline demos/TicTacToe/rebuild_demo.py:62]: Perhaps output the directory that you're assuming should exist?

File: demos/TicTacToe/run_demo.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  # Sets up the logging format and handler.
+  logger = logger_setup()
+ 
+  # Directory paths.
+  game_name = "tic_tac_toe_demo"
+  executable_dir = os.path.join(ROOT_DIRECTORY, game_name, "build", "bin", game_name, "Debug")
+
+  # Checks whether the demo was set up properly.
+  if not os.path.isdir(executable_dir):
+    logger.error("demo setup is incomplete.")
```
Comment (DellaBitta) [inline demos/TicTacToe/run_demo.py:58]: Again, output the directory you're looking for.  Maybe ask "Did you run the build script first?"

File: demos/TicTacToe/setup_demo.py [hunk 4]
```diff
@@ -59,35 +59,40 @@ def main():
...
-  log_run(ROOT_DIRECTORY, logger,"cocos new {0} -p com.DemoApp.{0} -l cpp -d .".format(game_name))
+  log_run(ROOT_DIRECTORY, logger, "cocos new {0} -p com.DemoApp.{0} -l cpp -d .".format(game_name))
 
   # Removing the default cocos2d-x project files.
-  log_run(ROOT_DIRECTORY, logger,"rm -r {0}/Classes {0}/Resources {0}/CMakeLists.txt".format(game_files_dir) )
+  log_run(ROOT_DIRECTORY, logger, "rm -r {0}/Classes {0}/Resources {0}/CMakeLists.txt".format(game_files_dir))
 
   # Copies the google-services.json file into the correct directory to run the executable. 
-  log_run(ROOT_DIRECTORY, logger,"cp google_services/google-services.json {}".format(os.path.join(game_resources_dir,"build","bin", game_name, "Debug")))
+  log_run(ROOT_DIRECTORY, logger, "cp google_services/google-services.json {}".format(os.path.join(game_resources_dir, "build", "bin", game_name, "Debug")))
```
Comment (DellaBitta) [inline demos/TicTacToe/setup_demo.py:81]: Can these be split up unto multiple lines? They surely must exceed the 80 character limit, no?

Comment (Grant-Postma) [reply demos/TicTacToe/setup_demo.py:81]: I went ahead and installed the linting and formatting suggested in the https://google.github.io/styleguide/pyguide.html. So the following commit will address this and all of the other formatting issues that are present in this iteration.

**Pull request merged**